### PR TITLE
SITES-206: Favor SSO authentication over local

### PR DESCRIPTION
**Pull request by jbickar**
# READY FOR REVIEW

# Summary
- Favor SSO authentication over local authentication.
- Favor SAML over WebAuth
- Allow Webauth if no SAML
- Allow local if no SAML nor WebAuth

# Needed By (Date)
- When does this need to be merged by? Socorro go-live

# Criticality
- How critical is this PR on a 1-10 scale? 9
- It has security and UX implications for going live with new sites on ACSF

# Steps to Test

1. Check out this branch
2. Install {{stanford_ssp}}
3. Go to the homepage
4. Ensure that "Log in with your SUNetID" links to `sso/login`
5. Disable {{stanford_ssp}} and {{stanford_simplesamlphp_auth}}
6. Enable {{webauth}}
7. Go to the homepage
8. Ensure that "Log in with your SUNetID" links to `sites/default/webauth/login`

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules? Sites 2.0/ACSF

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-206

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: templates/firststeps.tpl.php [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+}
+$user_login_text = l(t('Log in with your SUNetID'), 'sso/login') . '.';
+if ($sso_authentication_scheme == "webauth") {
+  $user_login_text = l(t('Log in with your SUNetID'), 'sites/default/webauth/login') . '.';
+}
 $user_login_block = "<h2>Welcome to Your Stanford Sites Website</h2>";
 // check if WMD is set to allow local Drupal logins.
 // if so, display the login text to have the user log in as user 1
-if ($webauth_allow_local === 0) {
+if ($$sso_authentication_scheme !== 'local') {
```
Comment (sherakama) [inline templates/firststeps.tpl.php]: Double dollar all the waaaaayyy

File: templates/firststeps.tpl.php [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
+/**
+ * Variables.
  */
-$user_login_text = l(t('Log in with your SUNetID.'), 'sites/default/webauth/login');
-$webauth_allow_local = variable_get('webauth_allow_local');
-$password_reset = l(t('password reset form'), 'user/password');
+
+// Authentication scheme defaults to SAML.
+$sso_authentication_scheme = 'saml';
+if (module_exists('webauth') && !module_exists('stanford_simplesamlphp_auth')) {
```
Comment (sherakama) [inline templates/firststeps.tpl.php:13]: I don't think you need to check for both. If someone has both enabled they have more problems than a missing login block.

Comment (sherakama) [reply templates/firststeps.tpl.php:13]: Belt and suspenders approach.

File: templates/firststeps.tpl.php [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+
+  $sso_authentication_scheme = 'webauth';
+}
+elseif (!module_exists('webauth') && !module_exists('stanford_simplesamlphp_auth')) {
+  $sso_authentication_scheme = 'local';
+}
+$user_login_text = l(t('Log in with your SUNetID'), 'sso/login') . '.';
+if ($sso_authentication_scheme == "webauth") {
+  $user_login_text = l(t('Log in with your SUNetID'), 'sites/default/webauth/login') . '.';
```
Comment (sherakama) [inline templates/firststeps.tpl.php]: You are allowed punctuation in the t() function. No need to concatenate the period after.

Comment (jbickar) [reply templates/firststeps.tpl.php]: I don't want the period in the hyperlink.

File: templates/firststeps.tpl.php (hunk 1 above)
Comment (jbickar) [reply templates/firststeps.tpl.php]: dolla dolla bill, y'all.

File: templates/firststeps.tpl.php (hunk 2 above)
Comment (jbickar) [reply templates/firststeps.tpl.php:13]: They have both enabled before running `drush sspwmd`. Now, if this block is still showing at that point, I kinda don't curr, but yeah, belt and suspenders.

**Review by sherakama — APPROVED** [review]

**Pull request merged**